pkg/resources/kafka/kafka.go

```diff
@@ -1217,6 +1225,44 @@ func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName strin
 	return nodePort, nil
 }
 
+func (r *Reconciler) getK8sAssignedNodeAddress(brokerId int32, nodeAddressType string) (string, error) {
+	podList := &corev1.PodList{}
+	if err := r.Client.List(context.TODO(), podList,
+		client.InNamespace(r.KafkaCluster.Namespace),
+		client.MatchingLabels(apiutil.MergeLabels(apiutil.LabelsForKafka(r.KafkaCluster.Name), map[string]string{v1beta1.BrokerIdLabelKey: strconv.Itoa(int(brokerId))})),
+	); err != nil {
+		return "", err
+	}
+
+	switch {
+	case len(podList.Items) == 0:
+		return "", fmt.Errorf("could not find pod with matching brokerId: %d in namespace '%s'", brokerId, r.KafkaCluster.Namespace)
+	case len(podList.Items) > 1:
+		return "", fmt.Errorf("multiple pod found with brokerId: %d in namespace '%s'", brokerId, r.KafkaCluster.Namespace)
+	default:
+		return r.getK8sNodeIp(podList.Items[0].Spec.NodeName, nodeAddressType)
+	}
+}
+
+func (r *Reconciler) getK8sNodeIp(nodeName string, nodeAddressType string) (string, error) {
```

Comment:
```suggestion
func (r *Reconciler) getK8sNodeIP(nodeName string, nodeAddressType string) (string, error) {
```
According to this: https://github.com/golang/go/wiki/CodeReviewComments#initialisms